src/util.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Get the current totals.
+ *
+ * @param {Object} cart The cart object.
+ * @returns {Number} The totals of the cart object, if available, else the totals of the cart progress bar preview.
+ */
+export function getCurrentTotal( cart ) {
+	if (
+		undefined !== cart &&
+		undefined !== cart.cartTotals.total_price &&
+		undefined !== cart.cartTotals.currency_minor_unit
+	) {
+		const total_price = cart.cartTotals.total_price;
+		const minor_unit = cart.cartTotals.currency_minor_unit;
+
+		return total_price / Math.pow( 10, minor_unit );
+	}
+	return '';
+}
+
+/**
+ * Get the minor unit.
+ *
+ * @param {Object} cart The cart object.
+ * @returns {Number} The totals of the cart object, if available, else the totals of the cart progress bar preview.
+ */
+export function getMinorUnit( cart ) {
+	if (
+		undefined !== cart &&
+		undefined !== cart.cartTotals.currency_minor_unit
+	) {
+		return cart.cartTotals.currency_minor_unit;
+	}
+	return '';
+}
+
+/**
+ * Get the remaining value.
+ *
+ * @param {Number} freeShippingFrom The free shipping from value.
+ * @param {Number} currentTotal The current total value.
+ * @param {Number} minorUnit The minor unit value.
+ * @returns {Number} The difference between freeShippingFrom and currentTotal.
+ */
+export function getRemaining( freeShippingFrom, currentTotal, minorUnit ) {
+	const remaining = freeShippingFrom - currentTotal;
+	const power = Math.pow( 10, minorUnit );
+	const result = remaining.toFixed( minorUnit ) * power;
+
+	return result;
```

Comment:
What's the point of using power here?

Reply:
Similar as before, I need the remaining value as minor unit price. I'm using pow to calculate the minor unit price based on the number of decimals.